monGARS/core/cortex/curiosity_engine.py

```
@@ -1,114 +1,262 @@
+        query_norm = math.sqrt(sum(value * value for value in query_vector))
+        if query_norm == 0:
+            logger.debug(
+                "Vector similarity fallback due to zero-length query embedding"
+            )
+            return self._count_token_similarity(query_terms, history_candidates)
+
+        try:
+            history_vectors = await asyncio.gather(
+                *(self.embedding_system.encode(item) for item in history_candidates)
+            )
+        except Exception as exc:  # pragma: no cover - embedding optional
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)